user-info-provider/src/test/java/de/intension/mapper/oidc/RoleBasedUserInfoProviderMapperTest.java

```diff
@@ -0,0 +1,143 @@
+package de.intension.mapper.oidc;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.keycloak.models.ClientSessionContext;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.ProtocolMapperModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.protocol.oidc.mappers.OIDCAttributeMapperHelper;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.representations.IDToken;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import de.intension.api.UserInfoAttribute;
+import de.intension.mapper.user.UserInfoHelper;
+
+public class RoleBasedUserInfoProviderMapperTest
+{
+
+    private static final String       SUB          = "af3a88fc-d766-11ec-9d64-0242ac120002";
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    @BeforeAll
+    static void setupObjectMapper()
+    {
+        objectMapper.registerModule(new JavaTimeModule());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"LEHR", "LEHR,EXTERN"})
+    void should_map_first_name_to_claim(String roleName)
+        throws IOException
+    {
+        RoleBasedUserInfoProviderMapper mapper = new RoleBasedUserInfoProviderMapper();
+        IDToken idToken = new IDToken();
+        idToken.setSubject(SUB);
+        KeycloakSession session = mock(KeycloakSession.class);
+        ClientSessionContext context = mock(ClientSessionContext.class);
+        mapper.transformIDToken(idToken, createMapperModel(mapper, roleName, false), session, createUserModel("LEHR"), context);
+        String userInfo = (String)idToken.getOtherClaims().get("userNameCustom");
+        Assertions.assertNotNull(userInfo);
+        Assertions.assertEquals("testFirstName", userInfo);
+
+        idToken = new IDToken();
+        idToken.setSubject(SUB);
+        mapper.transformIDToken(idToken, createMapperModel(mapper, roleName, true), session, createUserModel("LEHR"), context);
+        userInfo = (String)idToken.getOtherClaims().get("userNameCustom");
+        Assertions.assertNull(userInfo);
```

Comment:
we should not mix different use cases here
the method is called "should_map_first_name...", but it doesn't do it in this case.

Either we move it to the other test method or we have our own method